src/framework/preferences/js/Enactors.js

```diff
@@ -427,4 +427,244 @@ var fluid_2_0_0 = fluid_2_0_0 || {};
+        },
+        invokers: {
+            applyMuteAudio: {
+                funcName: "fluid.prefs.enactor.muteAudio.applyMuteAudio",
+                args: ["{arguments}.0"]
+            }
+        },
+        modelListeners: {
+            muteAudio: {
+                listener: "{that}.applyMuteAudio",
+                args: ["{change}.value"]
+            }
+        }
+    });
+
+    // An array in which all the components that have been muted by this option are stored
+    var mutedComponents = [];
+
+    fluid.prefs.enactor.muteAudio.applyMuteAudio = function (value) {
+        $(document).ready(function () {
+            var audios = $("audio, video");
+            if (value) {
+                audios.each(function () {
+                    if (!$(this).prop("muted")) {
+                        $(this).prop("muted", true);
+                        mutedComponents.push(this);
+                    }
+                });
+            }
+            else {
+                audios.each(function () {
+                    var index = mutedComponents.indexOf(this);
+                    if (index > -1) {
+                        $(this).prop("muted", false);
+                        mutedComponents.splice(index, 1);
+                    }
+                });
+            }
+        });
+    };
+
+    /*******************************************************************************
+     * blueColorFilter
+     *
+     * Sets the intensity of the blue color of the page to the multiple provided.
+     *******************************************************************************/
+
+    // Note that the implementors need to provide the container for this view component
+    fluid.defaults("fluid.prefs.enactor.blueColorFilter", {
+        gradeNames: ["fluid.prefs.enactor", "fluid.viewComponent"],
+        preferenceMap: {
+            "fluid.prefs.blueColorFilter": {
+                "model.value": "default"
+            }
+        },
+        expander: {
+            func: "{that}.addMutationObserver"
+        },
+        invokers: {
+            applyFilter: {
+                funcName: "fluid.prefs.enactor.blueColorFilter.applyFilter",
+                args: ["{that}.model.value"]
+            },
+            addMutationObserver: {
+                funcName: "fluid.prefs.enactor.blueColorFilter.addMutationObserver",
+                args: ["{that}.applyFilter"]
+            }
+        },
+        modelListeners: {
+            value: {
+                listener: "{that}.applyFilter"
+            }
+        }
+    });
+
+    /*Used in fluid.prefs.enactor.blueColorFilter.applyFilter and fluid.prefs.enactor.blueColorFilter.addMutationObserver
+     had to be defined here in order to not be create every time the model is changed */
+    var initialColorsDictionary = [];
+    var noMatchFlag = true;
+    var changedDOMFlag = false;
+    var blueColorFilterValue;
+
+    fluid.prefs.enactor.blueColorFilter.applyFilter = function (times) {
```

Comment:
This function is rather long and unclear. It can be refactored into a few functions that achieve one step a time. For example, one function to determine no matching color, the other to handle the case of no matching etc. Also, add code comments for clarification. Thanks.